src/lib/services/client-metrics/metrics-service-v2.ts

```diff
@@ -81,20 +112,28 @@ export default class ClientMetricsServiceV2 {
                 ),
         );
 
-        this.logger.debug(`got metrics from ${clientIp}`);
+        const validatedToggleNames = await this.validToggleNames(toggleNames);
```

Comment:
I'd prefer functions to be verbs. Maybe `filterValidToggleNames`? Also the variable should be `validToggleNames`